src/main/java/ui/MenuControl.java

```diff
@@ -271,6 +272,20 @@ private void onBoardDelete(String boardName) {
             }
         }).collect(Collectors.toList());
     }
+    
+    public void switchBoard() {
+        if (prefs.getLastOpenBoard().isPresent() && prefs.getAllBoards().size() > 1) {
+            List<String> boardNames = new ArrayList<>(prefs.getAllBoards().keySet());
```

Comment:
Just the former should do it. What do you mean by changing every access?

It'd be good to add a non-GUI test for this too. Save and delete a few boards, then test that cycling returns stuff in the expected order.